source/base/parameter_handler.cc

```diff
@@ -1608,6 +1608,31 @@ bool ParameterHandler::read_input (std::istream &input,
                                    const std::string &filename,
                                    const std::string &last_line)
 {
+  try
+    {
+      parse_input(input, filename, last_line);
+      return true;
+    }
+  catch (const ExcIO &exc)
+    {
+      throw;
+    }
+  // This catch block more or less duplicates the old behavior of this function,
+  // which was to print something to stderr for every parsing error (which are
+  // now exceptions) and then return false.
+  catch (const ExceptionBase &exc)
+    {
+      std::cerr << exc.what() << std::endl;
+      return false;
+    }
```

Comment:
I did not see a warning when I compiled this.

I think this is okay since the function just consists of a try-catch block. The only block without a return statement just contains `throw`.

In case this function changes in the future, though, you are right. I'll move the 'return false;` down a level.